### PR TITLE
Rename user.reset_password.validity to .token_validity

### PR DESCRIPTION
**Pull request by inkhey**
related to [tracim/tracim#970](https://github.com/tracim/tracim/issues/970)
to do:

- [X]  rename `user.reset_password.validity` to `user.reset_password.token_validity`.
- [X]  support for legacy config file .
- [X]  update changelog about this change.

### CODE REVIEW

**Comment by lebouquetin** [comment]
TODO:

- accept old parameter with a deprecation warning
- create a ticket in order to remove this deprecation warning / old parameter support
- document the breaking change in the changelog

File: backend/cypress_test.ini [hunk 1]
```diff
@@ -91,7 +91,7 @@ auth_type = internal
 # User auth token validity in seconds (used to interfaces like web calendars)
 user.auth_token.validity = 604800
 # user reset_password token validity (default to 900s -> 15 minutes)
-user.reset_password.validity = 900
+user.reset_password.token_validity = 900
```
Comment (lebouquetin) [inline backend/cypress_test.ini]: I suggest to rename to `token_lifetime` which seems to be more explicit. @inkhey @buxx @PhilippeAccorsi what do you think?

File: backend/development.ini.sample [hunk 2]
```diff
@@ -148,7 +148,7 @@ auth_types = internal
 ## User auth token validity in seconds (used to interfaces like web calendars)
 user.auth_token.validity = 604800
 ## user reset_password token validity (default to 900s -> 15 minutes)
-user.reset_password.validity = 900
+user.reset_password.token_validity = 900
```
Comment (lebouquetin) [inline backend/development.ini.sample]: same suggestion : `token_lifetime`

File: backend/tracim_backend/config.py [hunk 3]
```diff
@@ -227,10 +227,24 @@ def __init__(self, settings: typing.Dict[str, typing.Any]):
...
+                self,
+                'user.reset_password.validity parameter is deprecated ! '
+                'please use user.reset_password.token_validity instead.'
+            )
+        reset_password_validity = settings.get(
+            'user.reset_password.token_validity',
+            None
+        )
+        defaut_reset_password_validity = '900'
+        self.USER_RESET_PASSWORD_TOKEN_VALIDITY = int(reset_password_validity or \
```
Comment (lebouquetin) [inline backend/tracim_backend/config.py]: `USER_RESET_PASSWORD_TOKEN_VALIDITY` => `USER_RESET_PASSWORD_TOKEN_LIFETIME`

File: changelog.md [hunk 4]
```diff
@@ -4,6 +4,11 @@
 
 - If user does not manually change the language, email notifications will stay in English. In order to get notifications in French, change language to English, then back to French (issue #1374)
 
+### Others Changes
+
+- rename 'user.reset_password.validity' to 'user.reset_password.token_validity' in config,
```
Comment (lebouquetin) [inline changelog.md]: same thing

File: backend/cypress_test.ini (hunk 1 above)
Comment (tracim) [reply backend/cypress_test.ini]: I prefer that also.

File: backend/development.ini.sample (hunk 2 above)
Comment (tracim) [reply backend/development.ini.sample]: same.

File: backend/tracim_backend/config.py (hunk 3 above)
Comment (tracim) [reply backend/tracim_backend/config.py]: same

File: changelog.md (hunk 4 above)
Comment (tracim) [reply changelog.md]: same.

File: backend/cypress_test.ini (hunk 1 above)
Comment (PhilippeAccorsi) [reply backend/cypress_test.ini]: I prefer that also.

File: backend/development.ini.sample (hunk 2 above)
Comment (PhilippeAccorsi) [reply backend/development.ini.sample]: I prefer that also.

File: backend/tracim_backend/config.py (hunk 3 above)
Comment (PhilippeAccorsi) [reply backend/tracim_backend/config.py]: I prefer that also.

File: changelog.md (hunk 4 above)
Comment (PhilippeAccorsi) [reply changelog.md]: I prefer that also.

File: backend/cypress_test.ini (hunk 1 above)
Comment (buxx) [reply backend/cypress_test.ini]: `token_lifetime` +1

Comment (inkhey) [reply backend/cypress_test.ini]: I agree.

Comment (inkhey) [reply backend/cypress_test.ini]: Change done.

File: backend/development.ini.sample (hunk 2 above)
Comment (inkhey) [reply backend/development.ini.sample]: done.

File: backend/tracim_backend/config.py (hunk 3 above)
Comment (inkhey) [reply backend/tracim_backend/config.py]: done

File: changelog.md (hunk 4 above)
Comment (inkhey) [reply changelog.md]: done.

**Pull request merged**